homeassistant/components/template/select.py

```diff
@@ -187,34 +206,38 @@ def __init__(
         config: dict,
     ) -> None:
         """Initialize the entity."""
-        super().__init__(hass, coordinator, config)
+        TriggerEntity.__init__(self, hass, coordinator, config)
+        AbstractTemplateSelect.__init__(self, config)
+
+        if CONF_STATE in config:
+            self._to_render_simple.append(CONF_STATE)
+
+        self._attr_name = name = self._rendered.get(CONF_NAME, DEFAULT_NAME)
```

Comment:
why is setting `attr_name` now needed?

Reply:
It's not necessarily needed but it's making it consistent with other platforms.  Ultimately I will be replacing this in a pending PR where name is set in the abstract class for both trigger and state based template entities.